diff --git a/docs/auditctl.8 b/docs/auditctl.8
index 7d27990..2648a98 100644
--- a/docs/auditctl.8
+++ b/docs/auditctl.8
@@ -257,6 +257,9 @@ Saved User ID. See getresuid(2) man page.
 .TP
 .B uid
 User ID. May be numeric or the user account name.
+.TP
+.B nametype
+audit_names record type. numberic, AUDIT_TYPE_UNKNOWN=0,AUDIT_TYPE_NORMAL=1,AUDIT_TYPE_PARENT=2,AUDIT_TYPE_CHILD_DELETE=3,AUDIT_TYPE_CHILD_CREATE=4. in syscall open, AUDIT_TYPE_CHILD_CREATE mean file real create.
 .RE
 .TP
 .BI \-k\  key
diff --git a/lib/fieldtab.h b/lib/fieldtab.h
index b597caf..de920e0 100644
--- a/lib/fieldtab.h
+++ b/lib/fieldtab.h
@@ -70,3 +70,5 @@ _S(AUDIT_ARG3,         "a3"           )
 _S(AUDIT_FILTERKEY,    "key"          )
 _S(AUDIT_EXE,          "exe"          )
 _S(AUDIT_SADDR_FAM,    "saddr_fam"    )
+
+_S(AUDIT_NAMETYPE,     "nametype"     )
diff --git a/lib/libaudit.h b/lib/libaudit.h
index b344201..a88e286 100644
--- a/lib/libaudit.h
+++ b/lib/libaudit.h
@@ -406,6 +406,10 @@ extern "C" {
 #define AUDIT_SADDR_FAM 113
 #endif
 
+#ifndef AUDIT_NAMETYPE
+#define AUDIT_NAMETYPE 198
+#endif
+
 #ifndef AUDIT_SESSIONID
 #define AUDIT_SESSIONID 25
 #endif
